source/type/data.cpp

```diff
@@ -101,22 +104,45 @@ bool Data::load(const std::string& filename,
         LOG4CPLUS_INFO(logger, boost::format("stopTimes : %d nb foot path : %d Nombre de stop points : %d")
                        % pt_data->nb_stop_times()
                        % pt_data->stop_point_connections.size()
-                       % pt_data->stop_points.size()
-            );
-        if (chaos_database) {
-            fill_disruption_from_database(*chaos_database, *pt_data, *meta, contributors);
-        }
-        build_raptor(raptor_cache_size);
+                       % pt_data->stop_points.size());
     } catch(const wrong_version& ex) {
         LOG4CPLUS_ERROR(logger, "Cannot load data: " << ex.what());
         last_load = false;
+        return last_load;
     } catch(const std::exception& ex) {
         LOG4CPLUS_ERROR(logger, "Data loading failed: " << ex.what());
         last_load = false;
+        return last_load;
     } catch(...) {
         LOG4CPLUS_ERROR(logger, "Data loading failed");
         last_load = false;
+        return last_load;
 	}
+
+    // Load Disruption (optionnal)
+    if (chaos_database) {
+        try {
+            fill_disruption_from_database(*chaos_database, *pt_data, *meta, contributors);
+        } catch(const std::exception& ex) {
+            LOG4CPLUS_WARN(logger, "Data disruptions loading failed: " << ex.what());
+            disruption_error = true;
+        } catch(...) {
+            LOG4CPLUS_WARN(logger, "Data disruptions loading failed");
+            disruption_error = true;
+        }
+    }
+
+    // Build Raptor Data
+    try {
+        build_raptor(raptor_cache_size);
```

Comment:
@kinnou02 if this fails and we loaded disruptions, should we try to reload clean base_schedule data (no disruptions) and see if raptor builds?
Not sure of that, it's a bit of a silent crash... But being resilient is often crashing silently.

Reply:
If some disruptions were applied but not all we need to reload the data as you said

Reply:
same idea as above